### PR TITLE
[LA64_DYNAREC] Add adc8, adc8c, adc16 and testadc

### PR DESCRIPTION
**Pull request by xiangzhai**
Hi,

Testcase `testadc` Passed on `la64_lbt` supported platform:
```
Dynarec for LoongArch with extension LSX LASX LBT_X86 PageSize:16384 Running on Loongson-3A5000-HV with 4 Cores
Will use time-based emulation for rdtsc, even if hardware counter are available
Will use Software counter measured at 1.0 GHz emulating 2.0 GHz
Params database has 14 entries
Box64 with Dynarec v0.3.1  built on Nov 25 2024 11:44:37
BOX64: Didn't detect 48bits of address space, considering it's 39bits
Counted 59 Env var
BOX64 LIB PATH: BOX64 BIN PATH: ./:bin/:/home/zhaixiang/.cargo/bin/:/usr/local/bin/:/usr/bin/:/bin/:/usr/local/games/:/usr/games/:/home/zhaixiang/rust-1.77.0-loongarch64-unknown-linux-gnu/cargo/bin/:/home/zhaixiang/rust-1.77.0-loongarch64-unknown-linux-gnu/rustc/bin/
Looking for /home/zhaixiang/testadc
Rename process to "testadc" 
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) libdl.so.2
Using native(wrapped) libutil.so.1
Using native(wrapped) librt.so.1
Using native(wrapped) libbsd.so.0
al=0xfe
dl=0x1
ax=0xfffe
dx=0x1
eax=0xfffffffe
edx=0x1
```

Also passed when disabled `la64_lbt`:
```
export BOX64_DYNAREC_LA64NOEXT=1

Dynarec for LoongArch  PageSize:16384 Running on Loongson-3A5000-HV with 4 Cores
Will use time-based emulation for rdtsc, even if hardware counter are available
Will use Software counter measured at 1.0 GHz emulating 2.0 GHz
Params database has 14 entries
Box64 with Dynarec v0.3.1  built on Nov 25 2024 11:44:37
BOX64: Didn't detect 48bits of address space, considering it's 39bits
Counted 59 Env var
BOX64 LIB PATH: BOX64 BIN PATH: ./:bin/:/home/zhaixiang/.cargo/bin/:/usr/local/bin/:/usr/bin/:/bin/:/usr/local/games/:/usr/games/:/home/zhaixiang/rust-1.77.0-loongarch64-unknown-linux-gnu/cargo/bin/:/home/zhaixiang/rust-1.77.0-loongarch64-unknown-linux-gnu/rustc/bin/
Looking for /home/zhaixiang/testadc
Rename process to "testadc" 
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) libdl.so.2
Using native(wrapped) libutil.so.1
Using native(wrapped) librt.so.1
Using native(wrapped) libbsd.so.0
al=0xfe
dl=0x1
ax=0xfffe
dx=0x1
eax=0xfffffffe
edx=0x1
```

Please review my patch.

Thanks,
Leslie Zhai

### CODE REVIEW

File: src/dynarec/la64/dynarec_la64_00.c [hunk 1]
```diff
@@ -186,6 +206,24 @@ uintptr_t dynarec64_00(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
             emit_adc32(dyn, ninst, rex, gd, ed, x3, x4, x5, x6);
             break;
+        case 0x14:
+            INST_NAME("ADC AL, Ib");
+            READFLAGS(X_CF);
+            SETFLAGS(X_ALL, SF_SET_PENDING);
+            u8 = F8;
+            ANDI(x1, xRAX, 0xff);
+            emit_adc8c(dyn, ninst, x1, u8, x3, x4, x5, x6);
+            ANDI(xRAX, xRAX, ~0xff);
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_00.c:216]: I'm afraid this is wrong and unnecessary. You can replace the `ANDI` + `OR` with a `BSTRINS_D`. (Please pay attention to the architecture differences when coping code.)

File: tests/testadc.c
Comment (ksco) [inline tests/testadc.c:1]: The test is too trivial to be added as a test.

**Review by ksco — COMMENTED** [review]
Thanks for the contribution, please format your code using `clang-format` with the .clang-format file we have in the repo. I would like all the LA64 DynaRec code to be formatted.

**Review by ptitSeb — APPROVED** [review]

File: src/dynarec/la64/dynarec_la64_f0.c [hunk 2]
```diff
@@ -402,6 +402,31 @@ uintptr_t dynarec64_F0(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
             break;
+        case 0x11:
+            INST_NAME("LOCK ADC Ed, Gd");
+            READFLAGS(X_CF);
+            SETFLAGS(X_ALL, SF_SET_PENDING);
+            nextop = F8;
+            GETGD;
+            SMDMB();
+            if (MODREG) {
+                ed = xRAX + (nextop & 7) + (rex.b << 3);
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_f0.c:413]: This is wrong too.

**Pull request merged**